### PR TITLE
Multiple metrics

### PR DESCRIPTION
**Pull request by hanwag**
Related Issues from Jira:
* [Axim -190](https://fetchai.atlassian.net/jira/software/projects/AXIM/boards/84?selectedIssue=AXIM-190)
* [Axim -191](https://fetchai.atlassian.net/jira/software/projects/AXIM/boards/84?selectedIssue=AXIM-191)

### CODE REVIEW

File: colearn_examples/grpc/mnist_grpc.py [hunk 1]
```diff
@@ -87,7 +91,7 @@ def prepare_data_loaders(location: str,
 @FactoryRegistry.register_model_architecture(model_tag, [dataloader_tag])
 def prepare_learner(data_loaders: Tuple[PrefetchDataset, PrefetchDataset, PrefetchDataset],
                     steps_per_epoch: int = 100,
-                    vote_batches: int = 10,
+                    vote_batches: int = 1,
```
Comment (lrahmani) [inline colearn_examples/grpc/mnist_grpc.py]: why change the default to 1?

File: colearn_examples/grpc/mnist_grpc.py [hunk 2]
```diff
@@ -112,19 +119,19 @@ def prepare_learner(data_loaders: Tuple[PrefetchDataset, PrefetchDataset, Prefet
...
+    model.compile(loss=loss, metrics=metric_list, optimizer=opt)
 
     learner = KerasLearner(
         model=model,
         train_loader=data_loaders[0],
         vote_loader=data_loaders[1],
         test_loader=data_loaders[2],
-        criterion="sparse_categorical_accuracy",
-        minimise_criterion=False,
+        criterion="loss",
```
Comment (lrahmani) [inline colearn_examples/grpc/mnist_grpc.py:133]: to avoid having to set both `metric_list` and `criterion` separately would it be reasonable to :
- always expect the vote criterion to be part of the metric_list?
- simply passing `metric_list[0]` as the `criterion`?

Even more, how about passing to whole `metric_list` to the `KerasLearner` as a substitute for `criterion`?

**Review by lrahmani — COMMENTED** [review]
LGTM.  Please check inline comments for further details

File: colearn_examples/grpc/mnist_grpc.py (hunk 1 above)
Comment (hanwag) [reply colearn_examples/grpc/mnist_grpc.py]: The vote_score and test_score are directly effected by that. Until now the model has been tested batch wise (this means that in this case the set was split into 10 sets) on the test and vote set. The score e.g. accuracy was then calculated over the mean of those 10 sets. the reason of having a test set (same is valid for the vote set) in this case is to evaluate the model on the whole data, and not split it again into multiple parts. If we set this variable to 1 - the metric calculation is done properly.

What happens exactly if i set it to 10:
* There are ten rounds of testing (on each e.g. the accuracy will get calculated)
* Each round includes of 32 samples (defaullt value of keras)
* Then it takes the mean accuracy of the 10 rounds -> final score
* But this means only 32*10 samples are considered and not all

My thought here would be to remove the variable from the `register_model_architecture` and set it in the `keras_learner` itself to avoid that it will be overwritten in the future?

File: colearn_examples/grpc/mnist_grpc.py (hunk 2 above)
Comment (hanwag) [reply colearn_examples/grpc/mnist_grpc.py:133]: Metrics and loss are two different things. The model should try to minimize or maximize the loss function. The metrics are then used for checking how well the model did on these metrics. There are cases when the loss function is a metric that's true. But I would really keep it separate to avoid confusion / mixture of those two aspects.

**Review by lrahmani — APPROVED** [review]
LGTM
